### PR TITLE
[DSS-297] Hero: Design Audit + New Variant

### PR DESCRIPTION
**Pull request by pixelflips**
## Description
Updates hero component to include new variant and related UI changes based on design audit.

- Adds new contained variant to support non-full-bleed images.
- Updates the color of the play icon (charcoal).
- Removed branding prop from React version. (Could find no usage)
- Various minor adjustments and fixes to sync with the latest design spec.
- Documentation updates for a more wide array of examples.
- Adds a link to the React component from docs.

## Screenshots

|  Before  |  After  |
|--------|--------|
|Screenshot 2023-03-14 at 11 09 15 AM|Screenshot 2023-03-14 at 11 08 47 AM|

## Testing in `sage-lib`

**Rails/Docs**

- Navigate to [Hero](http://localhost:4000/pages/component/hero?tab=preview) 
- Verify the updated design tweaks & the new contained variant

**React/Storybook**

- Navigate to [Hero](http://localhost:4100/?path=/story/sage-hero--default) 
- Verify the updated design tweaks & the new contained variant

## Testing in `kajabi-products`

1. (**MEDIUM**) Updates hero to match latest design spec and adds contained variant.
   - [ ] Dashboard
   - [ ] Creator Tools > Projects List

## Related
DSS-297

### CODE REVIEW

File: packages/sage-react/lib/Hero/Hero.jsx [hunk 1]
```diff
@@ -2,13 +2,14 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
 import { Button } from '../Button';
-import { HERO_SIZES } from './configs';
+import { HERO_SIZES, } from './configs';
```
Comment (QuintonJason) [inline packages/sage-react/lib/Hero/Hero.jsx:5]: Can you remove this `,`?

File: packages/sage-assets/lib/stylesheets/components/_hero.scss [hunk 2]
```diff
@@ -58,30 +50,34 @@ $-hero-play-icon-background-color-hover: rgba(sage-color(white, 100), 1);
   max-width: map-get($sage-containers, "xl");
 
   @media (min-width: sage-breakpoint(xxl-min)) {
-    min-height: $-hero-max-height;
+    max-height: $-hero-max-height;
   }
 }
 
+.sage-hero--large.sage-hero--contained {
+
```
Comment (QuintonJason) [inline packages/sage-assets/lib/stylesheets/components/_hero.scss:58]: Remove this line break

File: packages/sage-assets/lib/stylesheets/components/_hero.scss [hunk 3]
```diff
@@ -122,20 +123,22 @@ $-hero-play-icon-background-color-hover: rgba(sage-color(white, 100), 1);
...
+  max-height: $-hero-max-height;
+  object-fit: cover;
 
-  @media (min-width: sage-breakpoint(md-min)) {
-    height: 100%;
-    object-fit: cover;
+  .sage-hero--contained & {
+    object-fit: unset;
+    height: auto;
+    max-height: rem(235px);
```
Comment (QuintonJason) [inline packages/sage-assets/lib/stylesheets/components/_hero.scss:135]: Can you put this value in a var?

File: packages/sage-assets/lib/stylesheets/components/_hero.scss [hunk 4]
```diff
@@ -91,25 +87,30 @@ $-hero-play-icon-background-color-hover: rgba(sage-color(white, 100), 1);
...
   }
 
-  @media (max-width: sage-breakpoint(sm-max)) {
-    height: 0;
-    padding-top: $-hero-mobile-aspect-ratio;
-    border-top-left-radius: sage-spacing(sm);
-    border-bottom-right-radius: 0;
+  .sage-hero--contained & {
+    height: auto;
+    margin: rem(36px);
```
Comment (QuintonJason) [inline packages/sage-assets/lib/stylesheets/components/_hero.scss:111]: Can this one use the margin var from above?

**Review by QuintonJason — CHANGES_REQUESTED** [review]
This changes look great! I'm requesting the React side also be more aligned with respect to demos. On the Rails side we have a bunch:
- Default
- Contained
- w/ CTA
- Contained w/ CTA
- Borderless with Custom Background

Would you be able to add the following stories?
- Default
- Contained
- w/ CTA
- Borderless with Custom Background

File: packages/sage-react/lib/Hero/Hero.jsx (hunk 1 above)
Comment (pixelflips) [reply packages/sage-react/lib/Hero/Hero.jsx:5]: Done.

**Review by QuintonJason — APPROVED** [review]
Looks and works great!

File: packages/sage-assets/lib/stylesheets/components/_hero.scss [hunk 5]
```diff
@@ -56,32 +45,27 @@ $-hero-play-icon-background-color-hover: rgba(sage-color(white, 100), 1);
...
 .sage-hero--large {
   max-width: map-get($sage-containers, "xl");
-
-  @media (min-width: sage-breakpoint(xxl-min)) {
-    min-height: $-hero-max-height;
-  }
 }
 
 .sage-hero__title {
   @extend %t-sage-heading-4;
```
Comment (anechol) [inline packages/sage-assets/lib/stylesheets/components/_hero.scss:51]: Should this be heading-3?

File: packages/sage-assets/lib/stylesheets/components/_hero.scss [hunk 6]
```diff
@@ -56,32 +45,27 @@ $-hero-play-icon-background-color-hover: rgba(sage-color(white, 100), 1);
...
-
-  @media (min-width: sage-breakpoint(md-min)) {
-    margin-top: $-hero-margin-y;
+  @media (min-width: sage-breakpoint(lg-max)) {
+    align-self: self-end;
+    padding: sage-spacing(md) 0 0 0;
   }
 }
 
 .sage-hero__body {
```
Comment (anechol) [inline packages/sage-assets/lib/stylesheets/components/_hero.scss:61]: Can we update this to match the design spec? The font-weight says 450 in the design file for this text, but it should be `medium` as medium = book. It's a long story...

File: packages/sage-assets/lib/stylesheets/components/_hero.scss [hunk 7]
```diff
@@ -56,32 +45,27 @@ $-hero-play-icon-background-color-hover: rgba(sage-color(white, 100), 1);
...
   grid-area: content;
+  padding: 0 sage-spacing(md);
   margin-bottom: $-hero-margin-y;
```
Comment (anechol) [inline packages/sage-assets/lib/stylesheets/components/_hero.scss:64]: Should this be 24px?

File: packages/sage-react/lib/Hero/Hero.jsx [hunk 8]
```diff
@@ -34,35 +40,27 @@ export const Hero = ({
...
+      )}
+      <div className="sage-hero__body">
+        {description && (
+          <div className={`${SageClassnames.SPACERS.MD_BOTTOM} sage-hero__description`}>
+            {description}
+          </div>
+        )}
+        {footerActions && (
+          <div className="sage-hero__footer-actions">
+            <Button.Group gap="sm">
```
Comment (anechol) [inline packages/sage-react/lib/Hero/Hero.jsx:59]: Should this be md?

**Review by anechol — CHANGES_REQUESTED** [review]
Nice work here! Just added some comments for design spec stuff.

**Comment by pixelflips** [comment]
> Nice work here! Just added some comments for design spec stuff.

@anechol thanks so much for catching those! I was so focused on the new image/layout I missed some of those smaller details. It's a big yes to all of those and I've updated them. Thanks again!

File: packages/sage-assets/lib/stylesheets/components/_hero.scss (hunk 7 above)
Comment (anechol) [reply packages/sage-assets/lib/stylesheets/components/_hero.scss:64]: I'm still seeing 32px here

**Review by anechol — CHANGES_REQUESTED** [review]
Thanks for making those changes. Just a couple more small things. It looks like the button gap between the CTAs should be 24px over 16px

File: packages/sage-assets/lib/stylesheets/components/_hero.scss (hunk 7 above)
Comment (pixelflips) [reply packages/sage-assets/lib/stylesheets/components/_hero.scss:64]: @anechol should be good to go now. Thanks so much for catching those!

**Review by anechol — APPROVED** [review]
LGTM!! 🚀

**Pull request merged**